ginivision/src/main/java/net/gini/android/vision/noresults/NoResultsActivity.java

```diff
@@ -53,22 +61,42 @@ public void onBackToCameraPressed() {
     protected void onCreate(@Nullable final Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.gv_activity_noresults);
-
+        setTitle("");
+        readExtras();
         final ActionBar supportActionBar = getSupportActionBar();
         if (supportActionBar != null) {
             supportActionBar.setDisplayHomeAsUpEnabled(true);
             supportActionBar.setDisplayShowHomeEnabled(true);
         }
-
         if (savedInstanceState == null) {
-            NoResultsFragmentCompat noResultsFragment = NoResultsFragmentCompat.createInstance();
-            noResultsFragment.setArguments(getIntent().getExtras());
-            getSupportFragmentManager().beginTransaction()
-                    .add(R.id.gv_fragment_noresults, noResultsFragment)
-                    .commit();
+            initFragment();
+        }
+    }
+
+    private void readExtras() {
+        Bundle extras = getIntent().getExtras();
+        if (extras != null) {
+            mDocument = extras.getParcelable(EXTRA_IN_DOCUMENT);
+        }
+        checkRequiredExtras();
+    }
+
+    private void checkRequiredExtras() {
+        if (mDocument == null) {
+            throw new IllegalStateException(
+                    "NoResultsActivity requires a Document. Set it as an extra using the EXTRA_IN_DOCUMENT key.");
```

Comment:
Why this exception is thrown but not caught afterwards? Did not Android Studio complain when `checkRequiredExtras` was called?

Reply:
That's a RuntimeException and doesn't have to be catched. Shouldn't happen in production or if it does there is a bigger issue of why is there no document anymore. If clients use this Activity and don't pass in the Document they will see this exception.